pkg/reconciler/tenancy/clusterworkspace/clusterworkspace_controller.go

```diff
@@ -240,6 +241,35 @@ func (c *Controller) process(ctx context.Context, key string) error {
 	previous := obj
 	obj = obj.DeepCopy()
 
+	reconcileMetadata(obj)
+
+	// If the object being reconciled changed as a result, update it.
+	if !equality.Semantic.DeepEqual(previous.ObjectMeta, obj.ObjectMeta) {
```

Comment:
This might be a silly question but if we are working on labels what is the value of checking the entire ObjectMetas are equal?

Reply:
No real reason, certainly could change `reconcileMetadata` and this block to operate strictly on labels